src/test/java/core/basesyntax/RegistrationTest.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Test
+    public void test_storageUser_containsUser() {
+        User retrievedUser = storageDao.get("CoolGuy");
+        assertEquals(userTest, retrievedUser, "Storage does not contain user");
+    }
+
+    @Test
+    public void test_storageUser_returnNull_isOk() {
+        User retrievedUser = storageDao.get("Coco123");
+        assertNull(retrievedUser, "Storage should return null to un-existing user");
+    }
+
+    /* Tests in class User */
+
+    @Test
+    public void test_equals_sameUser_ok() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("user1");
+        user2.setPassword("password");
+        user2.setAge(20);
+
+        assertEquals(user1, user2);
+    }
+
+    @Test
+    public void test_equals_differentUsers_notOK() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("user2");
+        user2.setPassword("password");
+        user2.setAge(25);
+
+        assertNotEquals(user1, user2);
+    }
+
+    @Test
+    public void test_hashCode_equalUsers_equalHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("user1");
+        user2.setPassword("password");
+        user2.setAge(20);
+
+        assertEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_hashCode_differentUsers_differentHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("user2");
+        user2.setPassword("password");
+        user2.setAge(25);
+
+        assertNotEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_gettersAndSetters_workOk() {
+        User user = new User();
+        user.setLogin("user1");
+        user.setPassword("password");
+        user.setAge(20);
+        user.setId(1L);
+
+        assertEquals("user1", user.getLogin());
+        assertEquals("password", user.getPassword());
+        assertEquals(20, user.getAge());
+        assertEquals(1, user.getId());
+
+        user.setLogin("newUser");
+        user.setPassword("newPassword");
+        user.setAge(25);
+        user.setId(2L);
+
+        assertEquals("newUser", user.getLogin());
+        assertEquals("newPassword", user.getPassword());
+        assertEquals(25, user.getAge());
+        assertEquals(2, user.getId());
+    }
+
+    /* Tests in class RegistrationServiceImpl */
+
+    @Test
+    public void test_checking_user_OnExisting() {
```

Comment:
Let's stick to the rules: 
There are a lot of ways to name your test methods. The main point is that they should have informative names and be consistent along with other developers in your team. For this task use such convention: <methodUnderTest>_<state>_<expectedBehavior>; For example, if we are testing the method register with a null
user's age the test method name should be register_nullAge_notOk. notOk is because the test expects the register method to throw an exception.